core/src/main/java/io/questdb/griffin/engine/table/PrometheusMetricsRecordCursorFactory.java

```diff
@@ -0,0 +1,364 @@
+        cursor = new PrometheusMetricsCursor(record);
+    }
+
+    @Override
+    public void _close() {
+        record.close();
+        cursor.close();
+    }
+
+    @Override
+    public RecordCursor getCursor(SqlExecutionContext sqlExecutionContext) {
+        cursor.of(sqlExecutionContext.getCairoEngine().getMetrics());
+        cursor.toTop();
+        return cursor;
+    }
+
+    @Override
+    public boolean recordCursorSupportsRandomAccess() {
+        return false;
+    }
+
+    @Override
+    public void toPlan(PlanSink sink) {
+        sink.type("prometheus_metrics");
+    }
+
+    public static class PrometheusMetricsCursor implements RecordCursor {
+        PrometheusMetricsRecord record;
+        private int pos;
+        private MetricsRegistry registry;
+        private int size;
+        private int subLimit;
+        private int subPos;
+
+        public PrometheusMetricsCursor(PrometheusMetricsRecord record) {
+            this.record = record;
+        }
+
+        @Override
+        public void close() {
+        }
+
+        @Override
+        public Record getRecord() {
+            return record;
+        }
+
+        @Override
+        public Record getRecordB() {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public boolean hasNext() throws DataUnavailableException {
+            if (subPos > -1 && subPos < subLimit - 1) {
+                subPos++;
+                record.of(registry.getTarget(pos), subPos);
+                return true;
+            }
+
+            subPos = -1;
+            subLimit = -1;
+
+            pos++;
+            if (pos < size) {
+                try {
+                    int amount = record.of(registry.getTarget(pos));
+                    if (amount > 1) {
+                        subPos = 0;
+                        subLimit = amount;
+                    }
+                    return true;
+                } catch (UnsupportedOperationException e) {
+                    return hasNext();
+                }
+            }
+            return false;
+        }
+
+        public void of(Metrics metrics) {
+            of(metrics.getRegistry());
+        }
+
+        public void of(MetricsRegistry registry) {
+            this.registry = registry;
+            this.size = registry.getSize();
+            this.toTop();
+        }
+
+        @Override
+        public long preComputedStateSize() {
+            return 0;
+        }
+
+        @Override
+        public void recordAt(Record record, long atRowId) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public long size() throws DataUnavailableException {
+            return -1;
+        }
+
+        @Override
+        public void toTop() {
+            pos = -1;
+            subPos = -1;
+            subLimit = -1;
+            if (record.isClosed) {
+                Misc.free(record.sink);
```

Comment:
toTop must not reallocate, it is used to pass over the same data set and often called per-record in cross-join. Thus it must be considered on hot path